lab_4_retrieval_w_clustering/main.py

```diff
@@ -67,6 +79,12 @@ def vectorize(self, tokenized_document: list[str]) -> Vector:
         Returns:
             Vector: BM25 vector for document.
         """
+        if not isinstance(tokenized_document, list) or not tokenized_document \
+                or not all(isinstance(word, str) for word in tokenized_document):
+            raise ValueError
+        if not self._vocabulary:
+            return ()
+        return self._calculate_bm25(tokenized_document)
```

Comment:
if this method returns None raise Error

Reply:
no, you are calling this function twice (!) call it once and remember the output in variable

Reply:
try to use walrus operator